test/metabase/models/setting_test.clj

```diff
@@ -199,30 +199,27 @@
               #"Cannot initialize setting before the db is set up"
               (setting/get :test-setting-custom-init)))))))
 
-(def ^:private fancy-type
-  {:type :json
-   :setter :none
-   :init str
-   :bloop? true})
-
-(deftest inject-base-test
-  (testing "The interaction between different :type values and the remaining options"
-    (let [expand            #'setting/inject-base
-          explicit-settings {:bloop? false :blep? 'obviously}
-          with-basic-type   (assoc explicit-settings :type :json)
-          with-map          (assoc explicit-settings :base fancy-type)
-          with-var          (assoc explicit-settings :base (symbol ::fancy-type))]
-      (testing "There's no magic when using regular keyword types"
-        (is (= with-basic-type (expand with-basic-type))))
-      (let [expected {:type   :json
-                      :setter :none
-                      :init   str
-                      :bloop? false
-                      :blep?  'obviously}]
-        (testing "You can use explicit maps for the mixin behaviour"
-          (is (= expected (expand with-map))))
-        (testing "You can use var references for the mixing behaviour"
-          (is (= expected (expand with-var))))))))
+(def base-options
+  {:setter   :none
+   :type     :csv
+   :default "base-default"})
+
+(defsetting test-base-setting
+  "Setting to test the `:feature` property of settings. This only shows up in dev."
+  :visibility :internal
+  :default    "my,default"
+  base-options)
+
+(deftest defsetting-with-base-test
```

Comment:
can this be marked `^:parallel`?

Reply:
I don't see any reason why not. Taking this as a hint add this annotation where possible for tests that I create or edit - sure! 😄

Reply:
Hmm, I got a false positive from `deftest-check-parallel` due to referencing the setter function, even though it was quoted. 

For now I've worked around it by building the symbol programmatically, but perhaps there's a better way? 

I'm not sure if we can just trust any symbolic references to unsafe functions, but perhaps we could support safety annotations. Either some metadata that this function reads, or using `hooks/reg-finding!` instead of `error!` inside the linter.